### ISSUE TITLE
Local Playlist Storage

### ISSUE THREAD

**Issue by Sikksens** [MeoMix/StreamusChromeExtension#630]
Hi there,

I know this project is EOL but the extension is still working so well for me and just wanted to find out if
it's possible store and save playlists locally, I only really use the extension on one computer so won't need any sharing functionality.

I know it's saving some data to local storage but I'm unsure as to how I could get playlists saving too.

**Comment by MeoMix** [comment]
Hahaha. What the hell are you doing still using my software? <3 

Oh gosh. I think it is possible, but it's been a while. I know that originally Streamus only wrote to local and the DB was added later. Let me look at it for a bit.

**Comment by Sikksens** [comment]
Of course :P
There's literally nothing else out there with the same features, the search from chrome bar and radio are amazing!!

I thought maybe replacing all the old url's with the Backbone.LocalStorage adapter but not entirely sure exactly which ones though.

**Comment by MeoMix** [comment]
Yeah you're definitely on the right path with replacing the URLs with localStorage adapter.

If you take a look at `streamItems.js`

https://github.com/MeoMix/StreamusChromeExtension/blob/Development/src/js/background/collection/streamItems.js 

you'll see it's implemented using the localStorage adapter.

A starting point would be to mirror this setup with Playlists and PlaylistItems:

- https://github.com/MeoMix/StreamusChromeExtension/blob/Development/src/js/background/collection/playlists.js

- https://github.com/MeoMix/StreamusChromeExtension/blob/Development/src/js/background/collection/playlistItems.js

My only concern is that there might be server-side logic modifying things slightly as they're saved, but at a glance I don't see anything too crazy going on:

- https://github.com/MeoMix/StreamusServer/blob/development/Streamus/Controllers/PlaylistController.cs

- https://github.com/MeoMix/StreamusServer/blob/development/Streamus/Controllers/PlaylistItemController.cs

You might try starting there and, if unsuccessful, let me know what seems to be the issue? I don't have this running on my machine right now at all.

**Comment by sravan-bala** [comment]
Try UpNext. It is a good chrome extension as well which serves same purpose. , They dont display the entire video but displays a still image from video

**Comment by MeoMix** [comment]
@sravan-bala  Just to be clear, UpNext is also in violation of YouTube's TOS. It'll be taken down if/when it ever becomes popular enough to incur their wrath. Just trying to set expectations.

**Comment by sravan-bala** [comment]
Oh is it? cruel world man!
